packages/coinstac-ui/app/render/components/dashboard/dashboard.jsx

```diff
@@ -75,6 +77,30 @@ function Dashboard({
   const threads = get(threadsData, 'fetchAllThreads');
   const remoteRuns = get(userRunsData, 'fetchAllUserRuns');
 
+  useEffect(() => {
+    const runs = filter(remoteRuns, (run => !!run.endDate)).map((run) => {
+      const consortium = find(consortia, (consortium => consortium.id === run.consortiumId));
+      const clients = keys(run.clients).map(clientId => ({
+        id: clientId,
+        username: run.clients[clientId],
+      }));
+
+      return {
+        ...omit(run, ['clients', 'consortiumId', '__typename']),
+        consortium: pick(consortium, ['id', 'name']),
+        clients,
+      };
+    });
+
+    const data = {
+      userId: auth.user.id,
+      appDirectory: auth.appDirectory,
+      runs,
+    };
+
+    ipcRenderer.send('generate-provenance', data);
+  }, [remoteRuns]);
+
```

Comment:
This won't capture local runs, though I'm not sure I like this either way. This looks like it'll try to generate provenance every mount + remote change, I think it'll be better to just generate provenance on pipeline finish, which can be found in `main.js`